ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3122,7 +3117,19 @@ Seems much cleaner if each stmt is identified as a particular HiveOperation (whi
     }
     return lockComponents;
   }
-  
+
+  private static LockType getLockTypeFromStorageHandler(WriteEntity output, Table t) {
+    final HiveStorageHandler storageHandler = Preconditions.checkNotNull(t.getStorageHandler(),
+        "Non-native tables must have an instance of storage handler.");
+    LockType lockType = storageHandler.getLockType(output);
+    if (null == LockType.findByValue(lockType.getValue())) {
+      throw new IllegalArgumentException(String
+          .format("Lock type [%s] for Database.Table [%s.%s] is unknown", lockType, t.getDbName(),
```

Comment:
Optional nit: I see this was copied from somewhere else, but there really only has to be one argument here, t.getCompleteName()

Reply:
`getCompleteName()` uses `@` as separator char. I think without quoting db and table identifiers both separator chars (`.` and `@`) adds the same value since both of them can be part of the identifier.
In this case this is not an issue because we just add the name to an exception message and the message also contains the format.